### PR TITLE
scripts: Add SPIRV-Tools option to stats

### LINKED ISSUE #2141: How to go about scanning SPIR-V Tools for VUIDs
The 1.2.152 headers add almost 300 VUs for Built-ins, most (if not all) of which I assume should be implemented in `spirv-val`

I created an [initial PR](https://github.com/KhronosGroup/SPIRV-Tools/pull/3756) to label all the current VUs in effort to know where to add effort in the battle for more complete Vulkan targeted SPIR-V validation

This issue is to address `vk_validation_stats.py` as I am willing to add the logic, but there a few a ways to go about it and wanted 2nd thoughts before I go off and create a PR as I realize there might be some internal flows I don't want to break on people.

1. Getting SPIR-V tools to scan source
    - One thought is that the script could attempt to clone the source ToT, only cloning the master branch in effort to reduce overhead
    - Another idea is just like the `validusage.json` to pass the path to a SPIR-V tools
    - Regardless, I feel this should work even if a user of the script
2. Marking where it the VU is implemented
    - Main issue is the VUID itself doesn't describe if it will be implemented in SPIR-V Tools or VVL until found in source
    - I imagine in the `-summary` it would be nice to show how many Explicit VUIDs of the total checked/tested are in SPIR-V tools repo
    - For the `-html`,`-cvs`, `-text` option it would be worth adding a column to describe if it is implemented in the Validation Layers or SPIR-V Tools
3. vk_validation_error_messages.h
    - There was some extra logic that was added to the ruby scripts in Vulkan-Doc to handle the fact these VUIDs belong to non-enum/commands/structs. I imagine there will need to be some extra logic for this file as well

Overall, I don't think this is too invasive of a change as it is all additive in nature

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Closes #2141 
Marked as WIP until https://github.com/KhronosGroup/SPIRV-Tools/pull/3756 is merged

This allows you to go

```
python vk_validation_stats.py validusage.json -spirvtools SPIRV-Tools/
```

### CODE REVIEW

File: scripts/vk_validation_stats.py [hunk 1]
```diff
@@ -69,12 +71,17 @@
...
+# File with information around where to search for validation in SPIRV-Tools
+spirvtools_info_file = "validation_info.json"
+
 def printHelp():
     print ("Usage:")
     print ("  python vk_validation_stats.py <json_file>")
     print ("                                [ -c ]")
     print ("                                [ -todo ]")
     print ("                                [ -vuid <vuid_name> ]")
+    print ("                                [ -unassigned ]")
```
Comment (mark-lunarg) [inline scripts/vk_validation_stats.py:83]: 👍

**Comment by sfricke-samsung** [comment]
@mark-lunarg The SPIRV-Tools changes are merged into ToT and this script is ready to be used.

- You will need the 1.2.152 headers as those are where the new SPIR-V built-in VUIDs are even added (will show as 0 VUIDs with 151 headers otherwise)
- We [discussed](https://github.com/KhronosGroup/SPIRV-Tools/pull/3756#discussion_r479730047) the idea of having the file paths in SPIR-V Tools, but decided to simply the flow with just listing them here

**Comment by mark-lunarg** [comment]
@sfricke-samsung, yep, will get to it after 152 lands -- it had an unusual amount of VUID churn and so is taking a bit longer than usual.  And after seeing your note that all the SPIR-V VUIDs were in one file, I agree that having the path here is the best thing to do.

**Comment by sfricke-samsung** [comment]
@mark-lunarg Sounds good, there is really no rush for me for this change, I am going to focusing on closing the about 66% coverage cap (in terms of the new VUID labels added) currently in Built-Ins in SPIR-V Tools and just wanted a way to track it... this PR is more of my courtesy of not just dumping a new process on you without volunteering to add the code... so quite honestly don't feel afraid to just refactor this PR as much as you want to fit your flow

**Comment by mark-lunarg** [comment]
This seemed to work fine in my testing.  Note that known-good is still set to behind your changes to spirv-tools, but it seems harmless enough to push.  Thanks Spencer!

**Review by mark-lunarg — APPROVED** [review]

**Pull request merged**